pandas/tseries/offsets.py

```diff
@@ -579,9 +580,43 @@ class BusinessHourMixin(BusinessMixin):
 
     def __init__(self, start='09:00', end='17:00', offset=timedelta(0)):
         # must be validated here to equality check
-        start = liboffsets._validate_business_time(start)
+        if _iterable_not_string(start):
+            start = np.asarray(start)
+            if len(start) == 0:
+                raise ValueError('number of starting time cannot be 0')
+        else:
+            start = np.array([start])
+        if _iterable_not_string(end):
+            end = np.asarray(end)
+            if len(end) == 0:
+                raise ValueError('number of ending time cannot be 0')
+        else:
+            end = np.array([end])
+
+        vliboffsets = np.vectorize(liboffsets._validate_business_time)
+        start = vliboffsets(start)
+        end = vliboffsets(end)
+
+        # Validation of input
+        if len(start) != len(end):
+            raise ValueError('number of starting time and ending time '
+                             'must be the same')
+        num_openings = len(start)
+
+        # sort starting and ending time by starting time
+        index = np.argsort(start)
+        start = tuple(start[index])
+        end = tuple(end[index])
+
+        total_secs = 0
+        for i in range(num_openings):
+            total_secs += self._get_business_hours_by_sec(start[i], end[i])
+            total_secs += self._get_business_hours_by_sec(
+                end[i], start[(i + 1) % num_openings])
+        if total_secs != 24 * 60 * 60:
+            raise ValueError('invalid starting and ending time(s)')
```

Comment:
Can you make this error message a little more informative?

Reply:
Updated